### DISCUSSION TITLE
Using Jolt Physics in Godot with C++ Extension

### DISCUSSION THREAD

**Discussion by whiletrue111** [godot-jolt/godot-jolt#995] [Questions]
Hello,

I am building a game using C++, and I would like to use the Godot physics engine interface with the Jolt engine directly from C++. I prefer not to use GDScript.

Can I interact with the Jolt extension from my C++ extension, or should I use the physics engine provided by the engine as is? What options do I have for integrating Jolt with C++?

Thank you.

**Comment by mihe** [comment]
The primary function of this extension is to provide an alternative "backend" for the regular physics engine interface in Godot, so you don't need to do anything special from any language to make use of Jolt. Just switch to the `JoltPhysics3D` physics engine as mentioned in the installation instructions and all the regular physics nodes (like `RigidBody3D` and `StaticBody3D`) will use Jolt under the hood regardless of what language you use.

Note that this does however not apply when talking about the Jolt-specific joint nodes (`JoltHingeJoint3D`, etc.), but the only thing they give you that the default joints don't have is the ability to do soft limits. This is talked about in #632.
